libraries/AP_Math/examples/throttle_curve/throttle_curve.cpp

```diff
@@ -0,0 +1,108 @@
+//
+// Unit tests for the AP_Math throttle curve code
+//
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Math/AP_Math.h>
+
+void test_throttle_curve(float alpha);
+void setup();
+void loop();
+
+const AP_HAL::HAL& hal = AP_HAL::get_HAL();
```

Comment:
This shouldn't need the HAL at all.

Reply:
it uses hal.console to print usage info :)

Reply:
printf will work just as well :-)